src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,266 @@
+package frc.team2412.robot.Subsystems;
+
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.RobotMap;
+import frc.team2412.robot.Commands.IndexerCommands.IntakeBalls;
+import frc.team2412.robot.Commands.IndexerCommands.ProcessBalls;
+import frc.team2412.robot.Commands.IndexerCommands.ShootBalls;
+import frc.team2412.robot.Commands.IndexerCommands.SwitchBalls;
+import frc.team2412.robot.Subsystems.constants.IndexerConstants.IntakeDirection;
+
+public class IndexerSubsystem extends SubsystemBase {

```

Comment:
ok, make the logic more command based with conditional commands, this is really cluttered and i do not want to read this. you and parker can do this